### PR TITLE
Fix auto_profile_cell_count range

### PR DESCRIPTION
**Pull request by haslinghuis**
Fixes: #2992

### CODE REVIEW

**Comment by blckmn** [comment]
AUTOMERGE: (FAIL)
 - github identifies PR as mergeable -> FAIL
 - assigned to a milestone -> PASS
 - cooling off period lapsed -> FAIL
 - commit count less or equal to three -> PASS
 - `Don't merge` label NOT found -> PASS
 - at least one `RN:` label found -> PASS
 - `Tested` label found -> PASS
 - assigned to an approver -> PASS
 - approver count at least three -> FAIL

**Review by McGiverGim — CHANGES_REQUESTED** [review]
To me is complicated to know what is 0 and 1. Maybe change this to a select? With -1 and 0 with some text like "Switch" and "Stay" and the 1 to 8 elements with something like 1S to 8S? With this and some tooltip describing the values will be great: https://github.com/betaflight/betaflight/pull/7516

**Comment by haslinghuis** [comment]
Updated with suggestion from @McGiverGim

**Review by Asizon — DISMISSED** [review]

File: locales/en/messages.json
Comment (McGiverGim) [inline locales/en/messages.json:3721]:
```suggestion
        "description": "Disable cell count for this profile"
```

File: locales/en/messages.json
Comment (McGiverGim) [inline locales/en/messages.json:3717]:
```suggestion
        "description": "Use this profile if there no profiles matching cell count"
```

Comment (McGiverGim) [reply locales/en/messages.json:3717]: The description is used to give context to translators. I'm not too sure if this description can help. Is not a good tooltip than a context for the word.

**Review by McGiverGim — CHANGES_REQUESTED** [review]

File: locales/en/messages.json
Comment (haslinghuis) [reply locales/en/messages.json:3721]: Copy pasta error. Fixed.

File: locales/en/messages.json
Comment (haslinghuis) [reply locales/en/messages.json:3717]: Fixed.

**Review by McGiverGim — APPROVED** [review]
The descriptions can be better but I approve it.
Think that translators only see a word "switch" and the description. It's good give information about the context: one of the options for auto profile switching depending on the battery capacity.

**Review by blckmn — APPROVED** [review]

**Comment by RipperDrone** [comment]
Perfect, didn't even think so far. Perfect clarity now 👍❤️

**Comment by haslinghuis** [comment]
@RipperDrone in GUI `-1` was not selectable before this change.

**Comment by RipperDrone** [comment]
maybe - but it was a documented setting which I have now preset on my 50+ quads... PITA to change 'em all :-(. Why not just keep the numbering scheme of designators whilest displaying a more descriptive text in BFC which definitely is a good thing to do?

**Comment by McGiverGim** [comment]
You don't understand how this works. The description has nothing to do with the internal value. Check nightly and you will see

**Comment by RipperDrone** [comment]
> You don't understand how this works. The description has nothing to do with the internal value. Check nightly and you will see

sry, I'm not an experienced coder to be fully able to interpret the code. However, looking at your assignments in the code snippet below, I'm assuming there is a NEW meaning of -1 as a value now (?), therefore my interpretation was that the code now is NOT backwards compatible anymore to quads that have had a '-1' setting stored in their diffs (= fallback profile to be used in case there was no distinct matching cell count found in any other profile).

Can you pls confirm whether backwards compatibility is preserved or broken?

image

**Comment by haslinghuis** [comment]
This PR only fixes the bug that -1 value in UI was not selectable and added a switch and meaningful labels for the available range. Cli is using it's own interface to communicate with firmware and has not been changed.

**Comment by RipperDrone** [comment]
Thank you for explaining 👍😍

**Pull request merged**
